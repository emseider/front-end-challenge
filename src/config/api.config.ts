export const API_BASE_URL: string = import.meta.env.VITE_API_BASE_URL;

export const API_KEY: string = import.meta.env.VITE_API_KEY;

export const API_HEADERS = {
  apikey: API_KEY,
  'api-version': '1.0',
  'Content-Type': 'application/json',
};

export const API_CONFIG = {
  baseURL: API_BASE_URL,
  headers: API_HEADERS,
};

