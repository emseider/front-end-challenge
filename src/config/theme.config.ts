import { createTheme } from "@mui/material";

export const theme = createTheme({
    palette: {
        primary: { main: '#000000' },
        secondary: { main: '#f5f5f5' },
    },
    components: {
        MuiCssBaseline: {
            styleOverrides: {
                body: {
                    display: 'block'
                },
            },
        },
        MuiTypography: {
            styleOverrides: {
                h4: {
                    fontWeight: 'bold',
                },
            },
        },
    },
});