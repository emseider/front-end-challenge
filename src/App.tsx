import { ThemeProvider } from "@emotion/react";
import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
import { CssBaseline } from "@mui/material";
import { QueryClientProvider } from "@tanstack/react-query";

import AppRoutes from "./App.routes";
import { queryClient } from "./config/query.config";
import { theme } from "./config/theme.config";
import "./App.css";
import { LocalizationProvider } from "@mui/x-date-pickers";

function App() {
  return (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={theme}>
        <LocalizationProvider dateAdapter={AdapterDayjs}>
          <CssBaseline />
          <AppRoutes />
        </LocalizationProvider>
      </ThemeProvider>
    </QueryClientProvider>
  );
}

export default App;
