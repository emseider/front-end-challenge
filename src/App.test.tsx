import { render } from "@testing-library/react";
import { ThemeProvider } from "@emotion/react";
import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
import { CssBaseline } from "@mui/material";
import { QueryClientProvider } from "@tanstack/react-query";
import { LocalizationProvider } from "@mui/x-date-pickers";
import App from "./App";
import { queryClient } from "./config/query.config";
import { theme } from "./config/theme.config";

describe("App component", () => {
  it("renders without crashing", () => {
    render(
      <QueryClientProvider client={queryClient}>
        <ThemeProvider theme={theme}>
          <LocalizationProvider dateAdapter={AdapterDayjs}>
            <CssBaseline />
            <App />
          </LocalizationProvider>
        </ThemeProvider>
      </QueryClientProvider>
    );
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <QueryClientProvider client={queryClient}>
        <ThemeProvider theme={theme}>
          <LocalizationProvider dateAdapter={AdapterDayjs}>
            <CssBaseline />
            <App />
          </LocalizationProvider>
        </ThemeProvider>
      </QueryClientProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
