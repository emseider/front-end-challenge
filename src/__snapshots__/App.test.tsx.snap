// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    Unexpected Application Error!
  </h2>
  <h3
    style="font-style: italic;"
  >
    useNavigate() may be used only in the context of a &lt;Router&gt; component.
  </h3>
  <pre
    style="padding: 0.5rem; background-color: rgba(200, 200, 200, 0.5);"
  >
    Error: useNavigate() may be used only in the context of a &lt;Router&gt; component.
    at invariant (/Users/em/projects/front-end-challenge/node_modules/react-router/dist/development/index.js:327:11)
    at useNavigateUnstable (/Users/em/projects/front-end-challenge/node_modules/react-router/dist/development/index.js:4276:3)
    at useNavigate (/Users/em/projects/front-end-challenge/node_modules/react-router/dist/development/index.js:4273:46)
    at InvoiceListPage (/Users/em/projects/front-end-challenge/src/pages/InvoiceListPage/InvoiceList.page.tsx:31:20)
    at Object.react-stack-bottom-frame (/Users/em/projects/front-end-challenge/node_modules/react-dom/cjs/react-dom-client.development.js:22428:20)
    at renderWithHooks (/Users/em/projects/front-end-challenge/node_modules/react-dom/cjs/react-dom-client.development.js:5757:22)
    at updateFunctionComponent (/Users/em/projects/front-end-challenge/node_modules/react-dom/cjs/react-dom-client.development.js:8018:19)
    at beginWork (/Users/em/projects/front-end-challenge/node_modules/react-dom/cjs/react-dom-client.development.js:9683:18)
    at runWithFiberInDEV (/Users/em/projects/front-end-challenge/node_modules/react-dom/cjs/react-dom-client.development.js:543:16)
    at performUnitOfWork (/Users/em/projects/front-end-challenge/node_modules/react-dom/cjs/react-dom-client.development.js:15052:22)
  </pre>
  <p>
    ðŸ’¿ Hey developer ðŸ‘‹
  </p>
  <p>
    You can provide a way better UX than this when your app throws errors by providing your own 
    <code
      style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
    >
      ErrorBoundary
    </code>
     or 
    <code
      style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
    >
      errorElement
    </code>
     prop on your route.
  </p>
</DocumentFragment>
`;
